test/Closures/update-funcexpr.js

```diff
@@ -0,0 +1,57 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+if (this.WScript && this.WScript.LoadScriptFile) { // Check for running in ch
+    this.WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+}
+
+var tests = [
+    {
+        name: "Functions can overwrite themselves",
+        body: function () {
+            function foo1() {
+                foo1 = 42;

```

Comment:
hmm, I thought this are syntax or early reference error.

Reply:
or may be not.

---
In reply to: [181449110](https://github.com/Microsoft/ChakraCore/pull/4987#discussion_r181449110) [](ancestors = 181449110)